crates/sui/src/client_commands.rs

```diff
@@ -980,8 +981,8 @@ impl SuiClientCommands {
                 let (package_id, compiled_modules, dependencies, package_digest, upgrade_policy) =
                     upgrade_package(
                         client.read_api(),
-                        build_config,
-                        package_path,
+                        build_config.clone(),
+                        package_path.clone(),
```

Comment:
We need the `install_dir` and `lock_file` from `build_config` and `package_path`. I couldn't find a great way to bypass cloning. Fortunately, I don't think these are overly big structs, so I think it's okay!

Reply:
With the `BuildConfig`, I think you should be able to pass it by reference basically everywhere? For paths, the trick is to accept an `impl AsRef<Path>` everywhere, and that way you could pass in a `PathBuf`, `&Path`, `&PathBuf` and it will always do the right thing.

Reply:
Yeah, I tried `&build_config` but there's a bit of trouble when `install_dir` eventuall gets passed down to lockfile logic that expects a `PathBuf` to `extend`. Similar issue for a callee of `package_path` that takes it in an `Option`, not straight `impl AsRef<Path>`.

I _think_ these things can be fixed up, especially the `package_path` one. Happy to revisit after this PR but don't know that it's sensible to tackle them here.